src/test/js/shared/idam_helper.js

```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
+                }),
+            });
+
+            if (!response.ok) {
+                throw new Error('Failed to fetch token');
+            }
+
+            const tokenData = await response.json();
+            TestData.FUNCTIONAL_TEST_TOKEN = tokenData.access_token;
+            return tokenData.access_token; // Assuming token is in the 'access_token' field
+        } catch (error) {
+            console.trace('Error fetching token:', error);
+            throw error;
+        }
+    }
+
+    createServiceUsingTestingSupportService(serviceName, serviceClientSecret, roleId, token, scope = [], ssoProviders = '') {
+        let data;
+
+        if (roleId === '') {
+
+            data = {
+                clientId: serviceName,
+                clientSecret : serviceClientSecret,
+                description: serviceName,
+                hmctsAccess: {
+                    mfaRequired: false,
+                    selfRegistrationAllowed: true,
+                    postActivationRedirectUrl: TestData.SERVICE_REDIRECT_URI,
+                    onboardingRoleNames: ['auto-private-beta_role'],
+                    ssoProviders:ssoProviders
+                },
+                oauth2: {
+                    issuerOverride: false,
+                    grantTypes: [
+                        "authorization_code",
+                        "refresh_token",
+                        "password",
+                        "implicit",
+                        "client_credentials"
+                    ],
+                    scopes : scope,
+                    redirectUris: [TestData.SERVICE_REDIRECT_URI],
+                    accessTokenLifetime: "PT0S",
+                    refreshTokenLifetime: "PT0S"
+                }};
+
+        } else {
+
+            data = {
+                clientId: serviceName,
+                clientSecret : serviceClientSecret,
+                description: serviceName,
+                allowedRoles: [roleId, 'auto-admin_role'],
+                hmctsAccess: {
+                    mfaRequired: false,
+                    selfRegistrationAllowed: true,
+                    postActivationRedirectUrl: TestData.SERVICE_REDIRECT_URI,
+                    onboardingRoleNames: ['auto-private-beta_role'],
+                    ssoProviders:ssoProviders
+                },
+                oauth2: {
+                    issuerOverride: false,
+                    grantTypes: [
+                        "authorization_code",
+                        "refresh_token",
+                        "password",
+                        "implicit",
+                        "client_credentials"
+                    ],
+                    scopes: scope,
+                    redirectUris: [TestData.SERVICE_REDIRECT_URI],
+                    accessTokenLifetime: "PT0S",
+                    refreshTokenLifetime: "PT0S"
+                }
+            };
+
+        }
+        console.trace(JSON.stringify(data));
```

Comment:
if you're going to leave logging in then you probably need to add some TODOs to tidy this up later on.